tests/CompositeChat/CompositeChatTest.cpp

```diff
@@ -38,21 +38,21 @@ static void TestParser1(void)
 static void TestParser2(void)
 {
 	cCompositeChat Msg;
-	Msg.ParseText("@3Advanced stuff: @5overriding color codes and http://links.with/@4color-in-them handling");
+	Msg.ParseText("&3Advanced stuff: &5overriding color codes and http://links.with/&4color-in-them handling");
 	const auto & Parts = Msg.GetParts();
 	TEST_EQUAL(Parts.size(), 4);
 
 	TEST_TRUE(std::holds_alternative<cCompositeChat::TextPart>(Parts[0]));
-	TEST_EQUAL(std::get<cCompositeChat::TextPart>(Parts[0]).Style, "@3");
+	TEST_EQUAL(std::get<cCompositeChat::TextPart>(Parts[0]).Style, "3");
 
 	TEST_TRUE(std::holds_alternative<cCompositeChat::TextPart>(Parts[1]));
-	TEST_EQUAL(std::get<cCompositeChat::TextPart>(Parts[1]).Style, "@5");
+	TEST_EQUAL(std::get<cCompositeChat::TextPart>(Parts[1]).Style, "35");
 
 	TEST_TRUE(std::holds_alternative<cCompositeChat::UrlPart>(Parts[2]));
-	TEST_EQUAL(std::get<cCompositeChat::UrlPart>(Parts[2]).Style, "@5");
+	TEST_EQUAL(std::get<cCompositeChat::UrlPart>(Parts[2]).Style, "35");
 
 	TEST_TRUE(std::holds_alternative<cCompositeChat::TextPart>(Parts[3]));
-	TEST_EQUAL(std::get<cCompositeChat::TextPart>(Parts[3]).Style, "@5");
+	TEST_EQUAL(std::get<cCompositeChat::TextPart>(Parts[3]).Style, "35");
```

Comment:
because it return two colors code. they are all added until you have a &r which means that at this point the color is and 3 then it's 5 and so the style is 35

Reply:
Hm weird. But what color does the game display? And earlier if there is a new color code the color was overridden. Is this the same as vanilla?

Reply:
it display always the last of the stack. I did it this way because there is not only "color code" but also "formating code" and those can be combined. this is vanilla in this regard. actually I don't know how mojang handle this but I though a "stack" system would be the best ? every &x character add to the stack until one does &r which clear the stack.